pkg/mtu/mtu.go

```diff
@@ -96,13 +98,17 @@ type Configuration struct {
 // specified, otherwise it will be automatically detected. if encapEnabled is
 // true, the MTU is adjusted to account for encapsulation overhead for all
 // routes involved in node to node communication.
-func NewConfiguration(authKeySize int, encryptEnabled bool, encapEnabled bool, mtu int) Configuration {
+func NewConfiguration(authKeySize int, encryptEnabled bool, encapEnabled bool, mtu int, nodeInternalIP net.IP) Configuration {
```

Comment:
nit: This variable is named `nodeInternalIP`, but the parameter is passed from `node.GetExternalIP()`. From the `mtu` package perspective there's no difference, this is just the `mtuDetectIP` so it would be less confusing to name it more neutrally (or at the very least, matching the callers).